File: app/src/main/java/com/doubean/ford/util/DateTimeStyle.java
package com.doubean.ford.util;

import java.time.LocalDateTime;

public enum DateTimeStyle implements IDateTimeStyle {
    SHORT {
        @Override
        public String getPattern(LocalDateTime localDateTime) {
            LocalDateTime now = LocalDateTime.now();
            if (now.getYear() == localDateTime.getYear()) {
                if (now.getMonth() == localDateTime.getMonth()) {
                    if (now.getDayOfMonth() == localDateTime.getDayOfMonth()) {
                        return "HH:mm";
                    } else {
                        return "MM-dd";
                    }
                } else {
                    return "MM-dd";
                }

            } else {
                return "yyyy-MM";
            }
        }
    },
    NORMAL {
        @Override
        public String getPattern(LocalDateTime localDateTime) {
            LocalDateTime now = LocalDateTime.now();
            if (now.getYear() == localDateTime.getYear()) {
                if (now.getMonth() == localDateTime.getMonth()) {
                    if (now.getDayOfMonth() == localDateTime.getDayOfMonth()) {
                        return "HH:mm";
                    } else {
                        return "MM-dd HH:mm";
                    }
                } else {
                    return "MM-dd HH:mm";
                }
            } else {
                return "yy-MM-dd HH:mm";
            }
        }
    },

    LONG {
        @Override
        public String getPattern(LocalDateTime localDateTime) {
            return "yyyy-MM-dd HH:mm";
        }
    }

}

File: app/src/main/java/com/doubean/ford/util/DateTimeStyle.kt
package com.doubean.ford.util

import java.time.LocalDateTime

enum class DateTimeStyle : IDateTimeStyle {
    SHORT {
        override fun getPattern(localDateTime: LocalDateTime): String {
            val now = LocalDateTime.now()
            return if (now.year == localDateTime.year) {
                if (now.month == localDateTime.month) {
                    if (now.dayOfMonth == localDateTime.dayOfMonth) {
                        "HH:mm"
                    } else {
                        "MM-dd"
                    }
                } else {
                    "MM-dd"
                }
            } else {
                "yyyy-MM"
            }
        }
    },
    NORMAL {
        override fun getPattern(localDateTime: LocalDateTime): String {
            val now = LocalDateTime.now()
            return if (now.year == localDateTime.year) {
                if (now.month == localDateTime.month) {
                    if (now.dayOfMonth == localDateTime.dayOfMonth) {
                        "HH:mm"
                    } else {
                        "MM-dd HH:mm"
                    }
                } else {
                    "MM-dd HH:mm"
                }
            } else {
                "yy-MM-dd HH:mm"
            }
        }
    },
    LONG {
        override fun getPattern(localDateTime: LocalDateTime): String {
            return "yyyy-MM-dd HH:mm"
        }
    }
}